### ISSUE TITLE
Modifying training for captioning

### ISSUE THREAD

**Issue by kushalj001** [bentrevett/pytorch-seq2seq#92]
I am trying to train a video captioning model based on the seq2seq tutorials. For this use case, my input would be a sequence of video frames instead of text.
Each video in the dataset has 20 captions given by different annotators. So, for each video, I need to loop again through its captions and then update the loss. My train method looks like this. Is this the correct way to go about it? 
@bentrevett it would be great if you could verify this. Thank you.
```
def train(model, loader, opt, criterion, clip):
    
    model.train()
    epoch_loss = 0.0
    
    for frames, captions in loader:
        
        # frames = [bs, seq_len, 3, 224, 224] = [bs, 32, 3, 224, 224]
        # captions = [bs, num_captions, seq_len] = [bs, 20, 20]
        
        captions = captions.permute(1, 2, 0).type(torch.LongTensor)
         # captions = [num_captions, seq_len, bs] = [20, 20, bs]
        
        frames = frames.to(device)
        captions = captions.to(device)
       
        
        for i in range(captions.shape[0]):
            
            opt.zero_grad()
            
            output = model(frames, captions[i])
            
            # output [trg_len, bs, output_vocab_dim]
            # captions[i] = [trg_len, bs]
            
            output_dim = output.shape[-1]
            
            output = output[1:].view(-1, output_dim)
            # output = [(trg_len-1)*bs, output_dim]
            
            target = captions[i][1:].view(-1)
            # target = [(trg_len-1)*bs]
            
            loss = criterion(output, target)
            
            loss.backward()
            
```

**Comment by bentrevett** [comment]
Looping through all of the captions and calculating the loss then updating is definitely what I'd try first. 

The only other option that comes to mind is by just picking a caption randomly and only doing a single update. 

Would be interesting to compare the two and see which is better.
